src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/PostTaskSearchControllerTest.java

```diff
@@ -708,13 +706,176 @@ public void should_return_a_200_with_search_results_and_correct_properties() {
             .body("tasks.auto_assigned", everyItem(notNullValue()))
             .body("tasks.warnings", everyItem(notNullValue()))
             .body("total_records", greaterThanOrEqualTo(1));
+    }
 
-        tasksCreated
-            .forEach(task -> common.cleanUpTask(task.getTaskId()));
+    @Test
+    public void should_return_every_items_have_same_work_type_when_search_by_work_type() {
+        setupForR2();
```

Comment:
Testing R2 logic should not be part of this class, there is new class for testing R2 for TaskSearch endpoint you can add it there.